code/modules/mob/language.dm

```diff
@@ -773,8 +773,13 @@
 	flags = RESTRICTED
 	syllables = list("BRAAAAAAAAAAAAAAAAINS", "BRAAINS", "BRAINS")
 
-/mob/proc/grant_all_languages()
+/mob/proc/grant_all_nonhivemind_languages()
 	for(var/la in GLOB.all_languages)
-		add_language(la)
+		var/datum/language/new_language = GLOB.all_languages[la]
+		if(!istype(new_language) || (new_language in languages))
```

Comment:
I don't think these sanity checks are necessary. The all_languages list only contains languages (if it doesn't, we should fix the cause instead of adding this sanity check) and the |= list addition already makes sure the mob's languages won't contain duplicates.